app/assets/javascripts/Components/peer_reviews_manager.jsx

```diff
@@ -0,0 +1,334 @@
+        actionString: action,
+        selectedReviewerGroupIds: reviewers,
+        selectedRevieweeGroupIds: reviewees,
+        selectedReviewerInRevieweeGroups: this.state.selectedReviewerInRevieweeGroups,
+        numGroupsToAssign: this.state.numReviewers
+      }
+    }).then(this.fetchData);
+  };
+
+  render(){
+    return (
+      <div>
+        <GradersActionBox
+          performAction={this.performButtonAction}
+          updateNumReviewers={this.updateNumReviewers}
+        />
+        <div className='mapping-tables'>
+          <div className='mapping-table'>
+            <ReviewersTable
+              ref={(r) => this.ReviewersTable = r}
+              groups={this.state.reviewerGroups}
+              reviewerToNumReviews={this.state.reviewerToNumReviews}
+              loading={this.state.loading} />
+          </div>
+          <div className='mapping-table'>
+            <RevieweesTable
+              ref={(r) => this.studentsTable = r}
+              groups={this.state.revieweeGroups}
+              revieweeToReviewers={this.state.revieweeToReviewers}
+              groupIdToName={this.state.groupIdToName}
+              onReviewerChangeInRevieweeTable={this.updatedSelectedReviewersInRevieweesTable}
+              selectedReviewerInRevieweeGroups={this.state.selectedReviewerInRevieweeGroups}
+              loading={this.state.loading}
+              showSections={this.props.showSections}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+
+class RawReviewersTable extends React.Component {
+  static columns = [
+    {
+      show: false,
+      accessor: '_id',
+      id: '_id'
+    },
+    {
+      Header: 'Reviewer Groups',
+      accessor: 'name',
+      id: 'name'
+    },
+    {
+      Header: 'Num Reviews',
+      accessor: 'groups',
+      className: 'groups',
+      filterable: false
+    }
+  ];
+
+  clearCheckboxes = () => {
+    this.refs.table.clearCheckboxes();
+  };
+
+  changeSection = (event) => {
+    this.clearCheckboxes();
+    this.setState({ sectionName: event.target.value });
+    this.props.refresh();
+  };
+
+  render() {
+    const hashmap = this.props.reviewerToNumReviews;
+    const groups_data = this.props.groups.map(function(group) {
+      let numReviews = 0;
+      if (hashmap.hasOwnProperty(group.id)) {
+        numReviews = hashmap[group.id];
+      }
+      return {
+        _id: group.id,
+        name: group.name,
+        groups: numReviews,
+        section: group.section
+      };
+    }.bind(this));
+
+    return (
+      <CheckboxTable
+        ref={(r) => this.checkboxTable = r}
+
+        data={groups_data}
+        columns={RawReviewersTable.columns}
+        defaultSorted={[
+          {
+            id: 'name'
+          }
+        ]}
+        loading={this.props.loading}
+        filterable
+
+        {...this.props.getCheckboxProps()}
+      />
+    );
+  }
+}
+
+
+class RawRevieweesTable extends React.Component {
+  getColumns = () => {
```

Comment:
This can be `static`. (The reason to use a function here is only if something in the column definitions requires `this`.